### PR TITLE
feat(core): add OKTA sso connector

### PR DESCRIPTION
**Pull request by simeng-li**
## Summary

Add OKTA SSO connector

image
image
image

## Testing

Test locally

## Checklist

- [ ] ~`.changeset`~
- [ ] ~unit tests~
- [ ] ~integration tests~
- [x] necessary TSDoc comments

### CODE REVIEW

**Review by darcyYe — APPROVED** [review]

**Review by wangsijie — APPROVED** [review]

File: packages/core/src/sso/OktaSsoConnector/index.ts [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+      ...conditional(email && email_verified && { email }),
+      ...conditional(phone && phone_verified && { phone }),
+      ...camelcaseKeys(rest),
+    };
+  }
+}
+
+export const oktaSsoConnectorFactory: SingleSignOnFactory<SsoProviderName.OKTA> = {
+  providerName: SsoProviderName.OKTA,
+  logo: 'https://logtodev.blob.core.windows.net/public-blobs/admin/r2a6qctI3lmG/2023/11/22/8bvg68e7/OKTA.D.png',
```
Comment (charIeszhao) [inline packages/core/src/sso/OktaSsoConnector/index.ts:57]: Will this address always be valid?

File: packages/core/src/sso/OktaSsoConnector/index.ts [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+import { type ExtendedSocialUserInfo } from '../types/saml.js';
+import { type SingleSignOnConnectorSession } from '../types/session.js';
+
+export class OktaSsoConnector extends OidcSsoConnector {
+  /**
+   * Override the getUserInfo method from the OidcSsoConnector class
+   *
+   * @remark Okta's IdToken does not include the sufficient user claims like email_verified, phone_verified, etc. {@link https://devforum.okta.com/t/email-verified-claim/3516/2}
+   * This method will fetch the user info from the userinfo endpoint instead.
+   */
```
Comment (charIeszhao) [inline packages/core/src/sso/OktaSsoConnector/index.ts:21]: Will there be sync issues between Okta and Logto userinfo endpoint?

File: packages/core/src/sso/OktaSsoConnector/index.ts (hunk 1 above)
Comment (simeng-li) [reply packages/core/src/sso/OktaSsoConnector/index.ts:57]: Offline synced. Will review these logo later.

File: packages/core/src/sso/OktaSsoConnector/index.ts (hunk 2 above)
Comment (simeng-li) [reply packages/core/src/sso/OktaSsoConnector/index.ts:21]: What do you mean by sync issues?

Comment (simeng-li) [reply packages/core/src/sso/OktaSsoConnector/index.ts:21]: Move forward for bug bash. Will offline sync.

**Pull request merged**